add-on/panels/test/panel_test.js

```diff
@@ -1475,4 +1478,129 @@ describe("loop.panel", function() {
         new sharedActions.DeleteRoom({ roomToken: roomData.roomToken }));
     });
   });
+
+  describe("SharePanelView", function() {
+    var view, dispatcher, roomStore;
+
+    function createTestComponent(extraProps) {
+      var props = _.extend({
+        dispatcher: dispatcher,
+        store: roomStore
+      }, extraProps);
+      return TestUtils.renderIntoDocument(
+        React.createElement(loop.panel.SharePanelView, props));
+    }
+
+    beforeEach(function() {
+      dispatcher = new loop.Dispatcher();
+      sandbox.stub(dispatcher, "dispatch");
+      roomStore = new loop.store.RoomStore(dispatcher, {
+        constants: {}
+      });
+
+      roomStore.setStoreState({
+        openedRoom: null,
+        pendingCreation: false,
+        pendingInitialRetrieval: false,
+        rooms: [],
+        error: undefined
+      });
+
+      view = createTestComponent();
+    });
+
+    it("should not open the panel if there is no room pending of creation", function() {
+      var panel = view.getDOMNode();
+
+      expect(panel.classList.contains("share-panel-open")).eql(false);
+    });
+
+    it("should open the panel after room creation", function() {
+      var panel = view.getDOMNode();
+
+      // Simulate that the user has click the browse button
+      roomStore.setStoreState({
+        pendingCreation: true
+      });
+
+      // Room has been created succesfully
+      roomStore.setStoreState({
+        pendingCreation: false
+      });
+
+      expect(view.state.shouldOpenSharePanel).eql(true);
+      expect(panel.classList.contains("share-panel-open")).eql(true);
+    });
+
+    it("should close the share panel when clicking the overlay", function() {
+      var overlay = view.getDOMNode().querySelector(".share-panel-overlay");
+      var panel = view.getDOMNode();
+
+      view.setState({
+        shouldOpenSharePanel: true
+      });
+
+      roomStore.setStoreState({
+        activeRoom: {
+          roomToken: "fakeToken"
+        }
+      });
+
+      TestUtils.Simulate.click(overlay);
+
+      expect(view.state.shouldOpenSharePanel).eql(false);
+      expect(panel.classList.contains("share-panel-open")).eql(false);
+      sinon.assert.calledOnce(dispatcher.dispatch);
+      sinon.assert.calledWithExactly(dispatcher.dispatch,
+        new sharedActions.OpenRoom({ roomToken: "fakeToken" }));
```

Comment:
The `sinon.assert`s here are duplicating the `OpenRoom` test. We should only need to test that there.